<!doctype html><html lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=robots content="noodp"><meta http-equiv=x-ua-compatible content="IE=edge"><title>ğŸ”– Biology in Julia - Los los los</title><meta name=Description content="Tech blog of Wen-Wei Tseng"><meta property="og:title" content="ğŸ”– Biology in Julia"><meta property="og:description" content="
Bioinformatics, genomics, agriculture, food science, medicine, genetic engineering, etc&mldr;1

See also

BioJulia
EcoJulia.
Julia Health
"><meta property="og:type" content="article"><meta property="og:url" content="https://sosiristseng.github.io/posts/julia/bookmark/bio/"><meta property="og:image" content="https://sosiristseng.github.io/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-04-23T10:41:40+00:00"><meta property="article:modified_time" content="2021-04-24T17:44:32+08:00"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://sosiristseng.github.io/"><meta name=twitter:title content="ğŸ”– Biology in Julia"><meta name=twitter:description content="
Bioinformatics, genomics, agriculture, food science, medicine, genetic engineering, etc&mldr;1

See also

BioJulia
EcoJulia.
Julia Health
"><meta name=application-name content="Los los los"><meta name=apple-mobile-web-app-title content="Los los los"><meta name=theme-color content="#ffffff"><meta name=msapplication-TileColor content="#da532c"><link rel=icon href=parrot.svg><link rel=apple-touch-icon sizes=180x180 href=/apple-touch-icon.png><link rel=mask-icon href=/safari-pinned-tab.svg color=#5bbad5><link rel=manifest href=/site.webmanifest><link rel=canonical href=https://sosiristseng.github.io/posts/julia/bookmark/bio/><link rel=prev href=https://sosiristseng.github.io/posts/julia/bookmark/audio-video/><link rel=next href=https://sosiristseng.github.io/posts/julia/bookmark/chem/><link rel=preload as=style onload="this.onload=null,this.rel='stylesheet'" href=https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.min.css><noscript><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.min.css></noscript><link rel=stylesheet href=/css/style.min.css><link rel=preload as=style onload="this.onload=null,this.rel='stylesheet'" href=https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.3/css/all.min.css><noscript><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.3/css/all.min.css></noscript><link rel=preload as=style onload="this.onload=null,this.rel='stylesheet'" href=https://cdn.jsdelivr.net/npm/animate.css@4.1.1/animate.min.css><noscript><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/animate.css@4.1.1/animate.min.css></noscript><script type=application/ld+json>{"@context":"http://schema.org","@type":"BlogPosting","headline":"ğŸ”– Biology in Julia","inLanguage":"en","mainEntityOfPage":{"@type":"WebPage","@id":"https:\/\/sosiristseng.github.io\/posts\/julia\/bookmark\/bio\/"},"genre":"posts","keywords":"julia, bookmark","wordcount":1264,"url":"https:\/\/sosiristseng.github.io\/posts\/julia\/bookmark\/bio\/","datePublished":"2021-04-23T10:41:40+00:00","dateModified":"2021-04-24T17:44:32+08:00","license":"This work is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License.","publisher":{"@type":"Organization","name":"xxxx"},"author":{"@type":"Person","name":"Wen-Wei Tseng"},"description":""}</script></head><body header-desktop=fixed header-mobile=auto><script type=text/javascript>if(window.localStorage&&localStorage.getItem('theme')){let a=localStorage.getItem('theme');a==='light'||a==='dark'||a==='black'?document.body.setAttribute('theme',a):window.matchMedia&&window.matchMedia('(prefers-color-scheme: dark)').matches?document.body.setAttribute('theme','dark'):document.body.setAttribute('theme','light')}else'auto'==='light'||'auto'==='dark'||'auto'==='black'?(document.body.setAttribute('auto'),window.localStorage&&localStorage.setItem('theme','auto')):(window.localStorage&&localStorage.setItem('theme','auto'),window.matchMedia&&window.matchMedia('(prefers-color-scheme: dark)').matches?document.body.setAttribute('theme','dark'):document.body.setAttribute('theme','light'));function getMeta(b){const a=document.getElementsByTagName('meta');for(let c=0;c<a.length;c++)if(a[c].getAttribute('name')===b)return a[c];return''}let themeColorMeta=getMeta('theme-color');document.body.getAttribute('theme')!='light'&&(themeColorMeta.content='#000000')</script><div id=mask></div><div class=wrapper><header class=desktop id=header-desktop><div class=header-wrapper><div class=header-title><a href=/ title="Los los los"><span class=header-title-pre><i class="fas fa-kiwi-bird fa-fw"></i></span>Los los los</a></div><div class=menu><div class=menu-inner><a class=menu-item href=/posts/ title=Posts>Posts </a><a class=menu-item href=/tags/ title=Tags>Tags </a><a class=menu-item href=/categories/ title=Categories>Categories </a><a class=menu-item href=/about/ title="About Me">About Me </a><a class=menu-item href=https://github.com/sosiristseng/ title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github-alt fa-fw"></i> </a><span class="menu-item delimiter"></span><span class="menu-item search" id=search-desktop><input type=text placeholder=Search id=search-input-desktop>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-desktop title=Search><i class="fas fa-search fa-fw"></i></a>
<a href=javascript:void(0); class="search-button search-clear" id=search-clear-desktop title=Clear><i class="fas fa-times-circle fa-fw"></i></a>
<span class="search-button search-loading" id=search-loading-desktop><i class="fas fa-spinner fa-fw fa-spin"></i></span>
</span><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme"><i class="fas fa-adjust fa-fw"></i></a></div></div></div></header><header class=mobile id=header-mobile><div class=header-container><div class=header-wrapper><div class=header-title><a href=/ title="Los los los"><span class=header-title-pre><i class="fas fa-kiwi-bird fa-fw"></i></span>Los los los</a></div><div class=menu-toggle id=menu-toggle-mobile><span></span><span></span><span></span></div></div><div class=menu id=menu-mobile><div class=search-wrapper><div class="search mobile" id=search-mobile><input type=text placeholder=Search id=search-input-mobile>
<a href=javascript:void(0); class="search-button search-toggle" id=search-toggle-mobile title=Search><i class="fas fa-search fa-fw"></i></a>
<a href=javascript:void(0); class="search-button search-clear" id=search-clear-mobile title=Clear><i class="fas fa-times-circle fa-fw"></i></a>
<span class="search-button search-loading" id=search-loading-mobile><i class="fas fa-spinner fa-fw fa-spin"></i></span></div><a href=javascript:void(0); class=search-cancel id=search-cancel-mobile>Cancel</a></div><a class=menu-item href=/posts/ title=Posts>Posts</a><a class=menu-item href=/tags/ title=Tags>Tags</a><a class=menu-item href=/categories/ title=Categories>Categories</a><a class=menu-item href=/about/ title="About Me">About Me</a><a class=menu-item href=https://github.com/sosiristseng/ title=GitHub rel="noopener noreffer" target=_blank><i class="fab fa-github-alt fa-fw"></i></a><a href=javascript:void(0); class="menu-item theme-switch" title="Switch Theme">
<i class="fas fa-adjust fa-fw"></i></a></div></div></header><div class="search-dropdown desktop"><div id=search-dropdown-desktop></div></div><div class="search-dropdown mobile"><div id=search-dropdown-mobile></div></div><main class=main><div class=container><div class=toc id=toc-auto><h2 class=toc-title>Contents</h2><div class=toc-content id=toc-content-auto></div></div><script>document.body.setAttribute("pageStyle","wide")</script><article class="page single"><h1 class="single-title animate__animated animate__flipInX">ğŸ”– Biology in Julia</h1><div class=post-meta><div class=post-meta-line><span class=post-author><a href=https://sosiristseng.github.io/ title=Author target=_blank rel="noopener noreffer author" class=author><i class="fas fa-user-circle fa-fw"></i>Wen-Wei Tseng</a></span>&nbsp;<span class=post-category>included in <a href=/categories/julia/><i class="far fa-folder fa-fw"></i>Julia</a>&nbsp;<a href=/categories/bookmark/><i class="far fa-folder fa-fw"></i>Bookmark</a></span></div><div class=post-meta-line><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime=2021-04-23>2021-04-23</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;1264 words&nbsp;
<i class="far fa-clock fa-fw"></i>&nbsp;6 minutes&nbsp;</div></div><div class="details toc" id=toc-static kept><div class="details-summary toc-title"><span>Contents</span>
<span><i class="details-icon fas fa-angle-right"></i></span></div><div class="details-content toc-content" id=toc-content-static><nav id=TableOfContents><ul><li><a href=#see-also>See also</a></li><li><a href=#api>API</a></li><li><a href=#modeling>Modeling</a></li><li><a href=#bioinformatics>Bioinformatics</a><ul><li><a href=#biojulia-packages>BioJulia packages</a></li><li><a href=#phylogeny>Phylogeny</a></li></ul></li><li><a href=#genome-wide-association-study-gwas>Genome Wide Association Study (GWAS)</a><ul><li><a href=#resources>Resources</a></li></ul></li><li><a href=#metabolic-networks>Metabolic networks</a></li><li><a href=#healthcare>Healthcare</a></li><li><a href=#biostatistics>BioStatistics</a></li><li><a href=#molecular-biology>Molecular Biology</a></li><li><a href=#microscopy>Microscopy</a></li><li><a href=#neuroscience>Neuroscience</a></li><li><a href=#pharmacology>Pharmacology</a></li><li><a href=#resources-and-teaching>Resources and Teaching</a></li></ul></nav></div></div><div class=content id=content><blockquote><p>Bioinformatics, genomics, agriculture, food science, medicine, genetic engineering, etc&mldr;<sup id=fnref:1><a href=#fn:1 class=footnote-ref role=doc-noteref>1</a></sup></p></blockquote><h2 id=see-also>See also</h2><ul><li><a href=https://biojulia.net/ target=_blank rel="noopener noreffer">BioJulia</a></li><li><a href=https://ecojulia.org target=_blank rel="noopener noreffer">EcoJulia</a>.</li><li><a href=https://juliahealth.org target=_blank rel="noopener noreffer">Julia Health</a></li></ul><ul><li>ğŸšï¸ means the package may not support current versions of Julia.</li><li>ğŸ—ï¸ means the package may be a WIP.</li></ul><h2 id=api>API</h2><p>APIs for biology and medicine databases.</p><ul><li><a href=https://github.com/BioJulia/BioServices.jl target=_blank rel="noopener noreffer">BioServices.jl</a> :: Julia interface to APIs for various bio-related web services.</li><li><a href=https://github.com/JuliaHealth/BioMedQuery.jl target=_blank rel="noopener noreffer">BioMedQuery.jl</a> :: Julia utilities to process and save results from BioMedical databases/APIs.</li><li><a href=https://github.com/JuliaHealth/PubChemCrawler.jl target=_blank rel="noopener noreffer">PubChemCrawler.jl</a> :: Utilities to programmatically query the PubChem database.</li><li><a href=https://github.com/JuliaHealth/PubMedMiner.jl target=_blank rel="noopener noreffer">PubMedMiner.jl</a> :: Julia package for studying co-occurrences in PubMed articles.</li></ul><h2 id=modeling>Modeling</h2><ul><li><a href=https://github.com/sdwfrost/Gillespie.jl target=_blank rel="noopener noreffer">Gillespie.jl</a> :: Stochastic Gillespie-type simulations using Julia.</li><li><a href=https://github.com/jangevaa/Pathogen.jl target=_blank rel="noopener noreffer">Pathogen.jl</a> :: Utilities to simulate and perform inference of disease dynamics.</li></ul><h2 id=bioinformatics>Bioinformatics</h2><p>As well as genomics.</p><ul><li><a href=https://github.com/farr/Ensemble.jl target=_blank rel="noopener noreffer">Ensemble.jl</a> :: Stochastic samplers based on the &ldquo;stretch move&rdquo; for ensembles of walkers.</li><li><a href=https://github.com/carlobaldassi/FastaIO.jl target=_blank rel="noopener noreffer">FastaIO.jl</a> :: Utilities to read/write FASTA format files in Julia.</li><li><a href=https://github.com/nw11/MethylUtils.jl target=_blank rel="noopener noreffer">MethylUtils.jl</a> :: Tools for WGBS Methylation analysis.</li><li><a href=https://github.com/diegozea/MIToS.jl target=_blank rel="noopener noreffer">MIToS.jl</a> :: Mutual Information Tools for protein Sequence analysis in Julia.</li><li><a href=https://github.com/BioJulia/MMTF.jl target=_blank rel="noopener noreffer">MMTF.jl</a> :: The Macromolecular Transmission Format (MMTF) is a new compact binary format to transmit and store biomolecular structures for fast 3D visualization and analysis.</li><li><a href=https://github.com/oheil/NormalizeQuantiles.jl target=_blank rel="noopener noreffer">NormalizeQuantiles.jl</a> :: implements quantile normalization</li><li><a href=https://github.com/pagnani/PlmDCA target=_blank rel="noopener noreffer">PlmDCA</a> :: Pseudo Likelihood Maximization for protein in Julia.</li><li><a href=https://github.com/Ward9250/SpeedDate.jl target=_blank rel="noopener noreffer">SpeedDate.jl</a> :: A utility for rapidly estimating coalescence times between sequences.</li><li><a href=https://github.com/reworkhow/XSim.jl target=_blank rel="noopener noreffer">XSim.jl</a> :: A fast and user-friendly tool to simulate sequence data and complicated pedigree structures.</li></ul><div class="details admonition warning"><div class="details-summary admonition-title"><i class="icon fas fa-exclamation-triangle fa-fw"></i>WIP or may not work<i class="details-icon fas fa-angle-right fa-fw"></i></div><div class=details-content><div class=admonition-content><ul><li>ğŸšï¸ <a href=https://github.com/Ward9250/BioArgParse.jl target=_blank rel="noopener noreffer">BioArgParse.jl</a> :: Extension to &lsquo;ArgParse.jl&rsquo;, enabling parsing of command line parameters to types in &lsquo;Bio.jl&rsquo;.</li><li>ğŸšï¸ <a href=https://github.com/nw11/BioFeatures.jl target=_blank rel="noopener noreffer">BioFeatures.jl</a>.</li><li>ğŸšï¸ <a href=https://github.com/nw11/BismarkSummary.jl target=_blank rel="noopener noreffer">BismarkSummary.jl</a> :: Basic summary of multiple Bismark runs.</li><li>ğŸšï¸ <a href=https://github.com/tanmaykm/Circuitscape.jl target=_blank rel="noopener noreffer">Circuitscape.jl</a> :: The Julia implementation of the <a href=http://www.circuitscape.org/ target=_blank rel="noopener noreffer">Circuitscape lib in Python</a> that uses electronic circuit theory to solve problems in landscape ecology.</li><li>ğŸšï¸ <a href=https://github.com/haploxer/COSMIC.jl target=_blank rel="noopener noreffer">COSMIC.jl</a> :: Data analysis engine for <a href=http://cancer.sanger.ac.uk/cosmic target=_blank rel="noopener noreffer">COSMIC</a> written in Julia.</li><li>ğŸšï¸ <a href=https://github.com/tlnagy/Crispulator.jl target=_blank rel="noopener noreffer">Crispulator.jl</a> :: A discrete simulation tool for designing pooled genetic screens.</li><li>ğŸšï¸ <a href=https://github.com/joefowler/DynamicTimeWarp.jl target=_blank rel="noopener noreffer">DynamicTimeWarp.jl</a> :: Implement Dynamic Time Warping for sequence alignment in Julia.</li><li>ğŸšï¸ <a href=https://github.com/seung-lab/EMIRT.jl target=_blank rel="noopener noreffer">EMIRT.jl</a> :: Electron Microscopy Image Reconstruction Toolbox using julia language.</li><li>ğŸšï¸ <a href=https://github.com/OpenGene/FusionDirect.jl target=_blank rel="noopener noreffer">FusionDirect.jl</a> :: Detect gene fusion directly from raw fastq files.</li><li>ğŸšï¸ <a href=https://github.com/nw11/GenomeAnnotations.jl target=_blank rel="noopener noreffer">GenomeAnnotations.jl</a> :: Manage Local Genome Annotation files.</li><li>ğŸšï¸ <a href=https://github.com/nw11/GenomicTiles.jl target=_blank rel="noopener noreffer">GenomicTiles.jl</a></li><li>ğŸšï¸ <a href=https://github.com/dcjones/gtf-parse-off target=_blank rel="noopener noreffer">gtf-parse-off</a> :: Experiments with parsing gene transfer format (GTF).</li><li>ğŸšï¸ <a href=https://github.com/kzahedi/HyperNEAT.jl target=_blank rel="noopener noreffer">HyperNEAT.jl</a> :: A generative encoding for evolving ANN based on the NeuroEvolution of Augmented Topologies (NEAT) algorithm for evolutionary computation.</li><li>ğŸšï¸ <a href=https://github.com/binarybana/OBC.jl target=_blank rel="noopener noreffer">OBC.jl</a> :: Optimal Bayesian classification for RNA-Seq data.</li><li>ğŸšï¸ <a href=https://github.com/OpenGene/OpenGene.jl target=_blank rel="noopener noreffer">OpenGene.jl</a> :: OpenGene core library in Julia.</li><li>ğŸšï¸ <a href=https://github.com/porterjamesj/Pagel.jl target=_blank rel="noopener noreffer">Pagel.jl</a> :: Detect correlated evolution on phylogenies.</li><li>ğŸšï¸ <a href=https://github.com/latticetower/ProgressiveAligner.jl target=_blank rel="noopener noreffer">ProgressiveAligner.jl</a> :: Progressive alignment scripts for protein sequences.</li><li>ğŸšï¸ <a href=https://github.com/nw11/PseudoGenomes.jl target=_blank rel="noopener noreffer">PseudoGenomes.jl</a> :: Read alleles without a VCF parser.</li><li>ğŸšï¸ <a href=https://github.com/slundberg/PureSeq.jl target=_blank rel="noopener noreffer">PureSeq.jl</a> :: Code for the Pure-seq project.</li><li>ğŸšï¸ <a href=https://github.com/PilgrimShadow/Rosalind.jl target=_blank rel="noopener noreffer">Rosalind.jl</a> :: A bioinformatics library for solving problems from rosalind.info.</li><li>ğŸšï¸ <a href=https://github.com/kdmurray91/seqhax.jl target=_blank rel="noopener noreffer">seqhax.jl</a> :: yet another NGS command.</li><li>ğŸšï¸ <a href=https://github.com/jamak/smoothlife-jl target=_blank rel="noopener noreffer">smoothlife-jl</a> :: an implementation of the SmoothLife algorithm in Julia.</li><li>ğŸšï¸ <a href=https://github.com/WestleyArgentum/Subsequences.jl target=_blank rel="noopener noreffer">Subsequences.jl</a> :: A package for finding longest common and longest contiguous subsequences.</li><li>ğŸšï¸ <a href=https://github.com/sckott/taxize.jl target=_blank rel="noopener noreffer">taxize.jl</a> :: A taxonomic toolbelt for Julia.</li><li>ğŸšï¸ <a href=https://github.com/tgvaughan/TimeTrees.jl target=_blank rel="noopener noreffer">TimeTrees.jl</a> :: Phylogenetic tree type for julia.</li><li>ğŸšï¸ <a href=https://github.com/mgvel/VariantCall.jl target=_blank rel="noopener noreffer">VariantCall.jl</a>.</li><li>ğŸšï¸ <a href=https://github.com/kzahedi/YARS.jl target=_blank rel="noopener noreffer">YARS.jl</a> :: YARS communication for RNA/proteins.</li></ul></div></div></div><h3 id=biojulia-packages>BioJulia packages</h3><ul><li><a href=https://github.com/BioJulia/BioAlignments.jl target=_blank rel="noopener noreffer">BioAlignments.jl</a> :: Sequence alignment tools in BioJulia</li><li><a href=https://github.com/BioJulia/BioSequences.jl target=_blank rel="noopener noreffer">BioSequences.jl</a> :: Biological sequences for the julia language.</li><li><a href=https://github.com/BioJulia/BioServices.jl target=_blank rel="noopener noreffer">BioServices.jl</a> :: Julia interface to APIs for various bio-related web services</li><li><a href=https://github.com/BioJulia/BioStructures.jl target=_blank rel="noopener noreffer">BioStructures.jl</a> :: A Julia package to read, write and manipulate macromolecular structures (particularly proteins).</li><li><a href=https://github.com/BioJulia/BioSymbols.jl target=_blank rel="noopener noreffer">BioSymbols.jl</a> :: Nucleic and amino acid primitive types.</li><li><a href=https://github.com/BioJulia/BioTools.jl target=_blank rel="noopener noreffer">BioTools.jl</a> :: Interfaces to common external biological tools from julia scripts and programs (e.g. BLAST).</li><li><a href=https://github.com/BioJulia/GeneticVariation.jl target=_blank rel="noopener noreffer">GeneticVariation.jl</a> :: types and methods for working with genetic variation.</li><li><a href=https://github.com/BioJulia/GenomeGraphs.jl target=_blank rel="noopener noreffer">GenomeGraphs.jl</a> :: A modern genomics framework for julia</li><li><a href=https://github.com/BioJulia/GenomicFeatures.jl target=_blank rel="noopener noreffer">GenomicFeatures.jl</a> :: utilities for working with interval based genomic annotations.</li><li><a href=https://github.com/BioJulia/IntervalTrees.jl target=_blank rel="noopener noreffer">IntervalTrees.jl</a> :: A data structure for efficient manipulation of sets of intervals.</li><li><a href=https://github.com/BioJulia/KmerAnalysis.jl target=_blank rel="noopener noreffer">KmerAnalysis.jl</a> :: Kmer counting algorithms and count-data utilities for the BioJulia framework</li></ul><h3 id=phylogeny>Phylogeny</h3><p>Evolution trees from biosequences.</p><ul><li><a href=https://github.com/BioJulia/Phylogenies.jl target=_blank rel="noopener noreffer">Phylogenies.jl</a> :: Phylogenetic trees and geneologies</li><li><a href=https://github.com/crsl4/PhyloNetworks.jl target=_blank rel="noopener noreffer">PhyloNetworks.jl</a> :: A Julia package for statistical inference, data manipulation and visualization of phylogenetic networks.</li><li><a href=https://github.com/jangevaare/PhyloTrees.jl target=_blank rel="noopener noreffer">PhyloTrees.jl</a> :: Phylogenetic tree simulation.</li></ul><h2 id=genome-wide-association-study-gwas>Genome Wide Association Study (GWAS)</h2><p><a href=https://en.wikipedia.org/wiki/Genome-wide_association_study target=_blank rel="noopener noreffer">ğŸ“– GWAS</a></p><ul><li><a href=https://github.com/reworkhow/JWAS.jl target=_blank rel="noopener noreffer">JWAS.jl</a> :: An open-source software tool written in Julia for Bayesian multiple regression methods applied to genome-wide association studies and genomic prediction.</li><li><a href=https://github.com/OpenMendel/MendelBase.jl target=_blank rel="noopener noreffer">MendelBase.jl</a> :: contains the base functions of OpenMendel.</li><li><a href=https://github.com/OpenMendel/SnpArrays.jl target=_blank rel="noopener noreffer">SnpArrays.jl</a> :: provides utilities for handling compressed storage of biallelic SNP (Single-Nucleotide Polymorphism) data.</li></ul><div class="details admonition warning"><div class="details-summary admonition-title"><i class="icon fas fa-exclamation-triangle fa-fw"></i>WIP or may not work<i class="details-icon fas fa-angle-right fa-fw"></i></div><div class=details-content><div class=admonition-content><ul><li>ğŸšï¸ <a href=https://github.com/DanielHoffmann32/AlignmentStatistics.jl target=_blank rel="noopener noreffer">AlignmentStatistics.jl</a> :: A package for statistical analyses of sequence alignments.</li><li>ğŸšï¸ <a href=https://github.com/dmbates/StatGenData.jl target=_blank rel="noopener noreffer">StatGenData.jl</a> :: Statistical analysis of genomic data.</li><li>ğŸšï¸ <a href=https://github.com/Tao-Hu/VarianceComponentTest.jl target=_blank rel="noopener noreffer">VarianceComponentTest.jl</a> :: A Julia package for performing exact variance component tests in genome-wide association study (GWAS).</li></ul></div></div></div><h3 id=resources>Resources</h3><ul><li>Wikipedia&rsquo;s <a href=http://en.wikipedia.org/wiki/List_of_RNA-Seq_bioinformatics_tools target=_blank rel="noopener noreffer">list of RNA-Seq bioinformatics tools</a>, not many of which are in Julia but depending on the language, they may have an API.</li><li><a href=https://genomicsclass.github.io/book/ target=_blank rel="noopener noreffer">Data Analysis for Genomics</a> : This is entirely in R-language but learners can use it to learn Julia.</li><li>ğŸšï¸ <a href=https://github.com/BioJulia/BioTutorials target=_blank rel="noopener noreffer">BioTutorials</a> : Tutorial Notebooks of BioJulia</li></ul><h2 id=metabolic-networks>Metabolic networks</h2><ul><li><a href=https://github.com/opencobra/COBRA.jl target=_blank rel="noopener noreffer">COBRA.jl</a> :: COnstraint-Based Reconstruction and Analysis, used to perform COBRA analyses such as Flux Balance Anlysis (FBA), Flux Variability Anlysis (FVA), or any of its associated variants such as <code>distributedFBA</code>.</li></ul><h2 id=healthcare>Healthcare</h2><p><a href=https://en.wikipedia.org/wiki/Category:Biomedicine target=_blank rel="noopener noreffer">ğŸ“– Biomedicine</a></p><ul><li><a href=https://github.com/JuliaHealth/DICOM.jl target=_blank rel="noopener noreffer">DICOM</a> :: Julia package for reading and writing DICOM (Digital Imaging and Communications in Medicine) files.</li><li><a href=https://github.com/INSIGNEO/openBF target=_blank rel="noopener noreffer">openBF</a> :: A finite-volume solver for elastic arterial blood flow networks.</li></ul><h2 id=biostatistics>BioStatistics</h2><ul><li><a href=https://github.com/diegozea/ROC.jl target=_blank rel="noopener noreffer">ROC.jl</a> :: Receiver Operating Characteristic (ROC) Curve for Julia Language.</li><li>ğŸšï¸ <a href=https://github.com/Rpedigree/pedigree.jl target=_blank rel="noopener noreffer">pedigree.jl</a> :: Pedigree extractors and representation as a Julia type with two integer members, sire and dam. A parent not in the pedigree is coded as 0.</li></ul><h2 id=molecular-biology>Molecular Biology</h2><ul><li>ğŸšï¸ <a href=https://github.com/timbitz/EvoDuplexes.jl target=_blank rel="noopener noreffer">EvoDuplexes.jl</a> :: A Julia package to fold all local and long-range RNA duplexes.</li><li><a href=https://github.com/bicycle1885/CellFishing.jl target=_blank rel="noopener noreffer">CellFishing.jl</a> :: (cell finder via hashing) is a tool to find similar cells of query cells based on their transcriptome expression profiles, a.k.a. single-cell sequencing.</li></ul><h2 id=microscopy>Microscopy</h2><p><a href=https://en.wikipedia.org/wiki/Microscopy target=_blank rel="noopener noreffer">ğŸ“– Microscopy</a></p><ul><li><a href=https://github.com/tlnagy/MicroscopyLabels.jl target=_blank rel="noopener noreffer">MicroscopyLabels.jl</a> :: Embed annotations in your microscopy images.</li></ul><h2 id=neuroscience>Neuroscience</h2><p><a href=http://codeneuro.org/ target=_blank rel="noopener noreffer">CodeNeuro</a> :: Bringing neuroscience and data science together.</p><ul><li><a href=https://github.com/sam81/BrainWave.jl target=_blank rel="noopener noreffer">BrainWave.jl</a> :: Julia functions to process electroencephalographic (EEG) recordings.</li><li><a href=https://github.com/davidssmith/DCEMRI.jl target=_blank rel="noopener noreffer">DCEMRI.jl</a> :: A Fast, Validated Toolkit for Dynamic Contrast Enhanced MRI Analysis. A paper on the code is in press at <a href=https://peerj.com/preprints/670/ target=_blank rel="noopener noreffer">PeerJ</a>.</li><li><a href=https://github.com/codles/EEG.jl target=_blank rel="noopener noreffer">EEG.jl</a> :: Process EEG files in Julia.</li><li><a href=https://github.com/babaq/NeuroAnalysis.jl target=_blank rel="noopener noreffer">NeuroAnalysis.jl</a> :: Julia package for neural signal analysis.</li><li><a href=https://github.com/Dhruva2/NeuronBuilder target=_blank rel="noopener noreffer">NeuronBuilder</a> :: Builds conductance based neural networks iteratively, from ion channels and synapses.</li><li><a href=https://github.com/galenlynch/OpenEphysLoader.jl target=_blank rel="noopener noreffer">OpenEphysLoader.jl</a> :: A set of tools to load data written by the Open Ephys GUI.</li></ul><div class="details admonition warning"><div class="details-summary admonition-title"><i class="icon fas fa-exclamation-triangle fa-fw"></i>WIP or may not work<i class="details-icon fas fa-angle-right fa-fw"></i></div><div class=details-content><div class=admonition-content><ul><li>ğŸšï¸ <a href=https://github.com/Andy-P/NEAT.jl target=_blank rel="noopener noreffer">NEAT.jl</a> :: Julia implemention of NEAT (NeuroEvolution of Augmenting Topologies) algorithm.</li><li>ğŸšï¸ <a href=https://github.com/simonster/SkullSegment.jl target=_blank rel="noopener noreffer">SkullSegment.jl</a> :: Macaque brain and skull extraction tools.</li><li>ğŸšï¸ <a href=https://github.com/grero/SpikeSorter.jl target=_blank rel="noopener noreffer">SpikeSorter.jl</a> :: Spike sorting tool for experimental neuroscientists.</li></ul></div></div></div><h2 id=pharmacology>Pharmacology</h2><p><a href=https://en.wikipedia.org/wiki/Pharmacology target=_blank rel="noopener noreffer">ğŸ“– Pharmacology</a></p><p>Also see <a href=https://github.com/PumasAI target=_blank rel="noopener noreffer">Pumas AI</a>, Pharmaceutical Modeling and Simulation in Julia.</p><ul><li><a href=https://github.com/PumasAI/DataInterpolations.jl target=_blank rel="noopener noreffer">DataInterpolations.jl</a> :: A library of data interpolation and smoothing functions.</li><li><a href=https://github.com/PumasAI/PumasTutorials.jl target=_blank rel="noopener noreffer">PumasTutorials.jl</a> :: Tutorials for pharmaceutical modeling and simulation with Pumas.jl. <a href=http://tutorials.pumas.ai>http://tutorials.pumas.ai</a></li></ul><h2 id=resources-and-teaching>Resources and Teaching</h2><blockquote><p>Some resources and teaching aids listed here are not Julia language specific but you may be able to find something common and useful that can be reused and shared with attribution if it is CC-licensed.</p></blockquote><ul><li>A google document listing <a href="https://docs.google.com/spreadsheets/d/1BjKsN0B1hqd4dJW5slZ5KPuToCjSMRyA7Bl8MwWrbS4/edit#gid=0" target=_blank rel="noopener noreffer">Python for biologists resources</a> by Lenny Teytelman (@lteytelman on twitter).</li><li><a href=https://software-carpentry.org/lessons.html target=_blank rel="noopener noreffer">Software Carpentry</a> teaching material.</li><li><a href=https://www.ebi.ac.uk/training/on-demand target=_blank rel="noopener noreffer">Train online @ EMBL-EBI</a> provides free courses on Europe&rsquo;s most widely used data resources, created by experts at EMBL-EBI and collaborating institutes.</li><li><a href=https://github.com/aidanbudd/trainingPhyloIntro target=_blank rel="noopener noreffer">TrainingPhyloIntro</a> :: Practical exercises and slides for teaching computational molecular evolution.</li><li><a href=https://github.com/aidanbudd/capetown2014 target=_blank rel="noopener noreffer">capetown2014</a> :: Teaching material for EMBO Practical Course on Computational analysis of protein-protein interactions - From sequences to networks run in Cape Town 2014.</li><li><a href=https://github.com/aidanbudd/course_EMBO_at_TGAC_PPI_Sep2015 target=_blank rel="noopener noreffer">course_EMBO_at_TGAC_PPI_Sep2015</a> :: Program and some material for EMBO Practical Course on computational analysis of protein-protein interactions: From sequences to networks, taking place at TGAC, Norwich, UK, Sep-Oct 2015.</li><li>Ian Quigley&rsquo;s (@mucociliary on twitter) notes on teaching <a href=http://daudin.icmb.utexas.edu/wiki/index.php/List_of_sessions target=_blank rel="noopener noreffer">applied bioinformatics / compbio</a>.</li><li><a href=http://ubiquitousgenomics.teamerlich.org target=_blank rel="noopener noreffer">Ubiquitous Genomics</a> :: COMSE6998/ Class 2015 by Prof. Yaniv Erlich. The <a href=https://github.com/erlichya/ubiquitousgenomics target=_blank rel="noopener noreffer">source is available on github</a>.</li></ul><section class=footnotes role=doc-endnotes><hr><ol><li id=fn:1 role=doc-endnote><p><a href=https://github.com/svaksha/Julia.jl target=_blank rel="noopener noreffer">Julia.jl</a> is under COPYRIGHT Â© 2012-Now SVAKSHA, dual-licensed for the data (ODbL-v1.0+) and the software (AGPLv3+), respectively.&#160;<a href=#fnref:1 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li></ol></section></div><div class=post-footer id=post-footer><div class=post-info><div class=post-info-line><div class=post-info-mod><span>Updated on 2021-04-24&nbsp;<a class=git-hash href=https://github.com/sosiristseng/sosiristseng.github.io/commit/5fc1422b1a0c4b9b2898c5a9c94b6c45cf9bf208 target=_blank title="commit by Wen-Wei Tseng(sosiristseng@gmail.com) 5fc1422b1a0c4b9b2898c5a9c94b6c45cf9bf208: Move outdated to hidden blocks Fix #203 Fix #202" rel="noopener noreferrer">
<i class="fas fa-hashtag fa-fw"></i>5fc1422</a></span></div><div class=post-info-license></div></div><div class=post-info-line><div class=post-info-md><span><a class=link-to-markdown href=/posts/julia/bookmark/bio/index.md target=_blank rel="noopener noreferrer">Read Markdown</a></span></div><div class=post-info-share><span><a href=javascript:void(0); title="Share on Twitter" data-sharer=twitter data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/ data-title="ğŸ”– Biology in Julia" data-via=sosiristseng data-hashtags=julia,bookmark><i class="fab fa-twitter fa-fw"></i></a><a href=javascript:void(0); title="Share on Facebook" data-sharer=facebook data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/ data-hashtag=julia><i class="fab fa-facebook-square fa-fw"></i></a><a href=javascript:void(0); title="Share on Linkedin" data-sharer=linkedin data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/><i class="fab fa-linkedin fa-fw"></i></a><a href=javascript:void(0); title="Share on Hacker News" data-sharer=hackernews data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/ data-title="ğŸ”– Biology in Julia"><i class="fab fa-hacker-news fa-fw"></i></a><a href=javascript:void(0); title="Share on Reddit" data-sharer=reddit data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/><i class="fab fa-reddit fa-fw"></i></a><a href=javascript:void(0); title="Share on Line" data-sharer=line data-url=https://sosiristseng.github.io/posts/julia/bookmark/bio/ data-title="ğŸ”– Biology in Julia"><i data-svg-src=https://cdn.jsdelivr.net/npm/simple-icons@4.23.0/icons/line.svg></i></a></span></div></div></div><div class=post-info-more><section class=post-tags><i class="fas fa-tags fa-fw"></i>&nbsp;<a href=/tags/julia/>julia</a>,&nbsp;<a href=/tags/bookmark/>bookmark</a></section><section><span><a href=javascript:void(0); onclick=window.history.back()>Back</a></span>&nbsp;|&nbsp;<span><a href=/>Home</a></span></section></div><div class=post-nav><a href=/posts/julia/bookmark/audio-video/ class=prev rel=prev title="ğŸ”– Audio and Video processing in Julia"><i class="fas fa-angle-left fa-fw"></i>ğŸ”– Audio and Video processing in Julia</a>
<a href=/posts/julia/bookmark/chem/ class=next rel=next title="ğŸ”– Chemistry in Julia">ğŸ”– Chemistry in Julia<i class="fas fa-angle-right fa-fw"></i></a></div></div></article></div></main><footer class=footer><div class=footer-container><div class=footer-line>Powered by <a href=https://gohugo.io/ target=_blank rel="noopener noreffer" title="Hugo 0.83.1">Hugo</a> | Theme - <a href=https://github.com/HEIGE-PCloud/DoIt target=_blank rel="noopener noreffer" title="DoIt 0.2.10"><i class="far fa-edit fa-fw"></i> DoIt</a></div><div class=footer-line><i class="far fa-copyright fa-fw"></i><span itemprop=copyrightYear>2020 - 2021</span><span class=author itemprop=copyrightHolder>&nbsp;<a href=https://sosiristseng.github.io/ target=_blank rel="noopener noreferrer">Wen-Wei Tseng</a></span>&nbsp;|&nbsp;<span class=license><a rel="license external nofollow noopener noreferrer" href=https://creativecommons.org/licenses/by-nc/4.0/ target=_blank>CC BY-NC 4.0</a></span></div></div></footer></div><div id=fixed-buttons><a href=# id=back-to-top class=fixed-button title="Back to Top"><i class="fas fa-arrow-up fa-fw"></i>
</a><a href=# id=view-comments class=fixed-button title="View Comments"><i class="fas fa-comment fa-fw"></i></a></div><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.css><script type=text/javascript src=https://cdn.jsdelivr.net/npm/autocomplete.js@0.38.0/dist/autocomplete.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/algoliasearch@4.9.1/dist/algoliasearch-lite.umd.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/lazysizes@5.3.2/lazysizes.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/clipboard@2.0.8/dist/clipboard.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/sharer.js@0.4.1/sharer.min.js></script><script type=text/javascript src=https://cdn.jsdelivr.net/npm/cookieconsent@3.1.1/build/cookieconsent.min.js></script><script type=text/javascript>window.config={code:{copyTitle:"Copy to clipboard",maxShownLines:14},comment:{},cookieconsent:{content:{dismiss:"Got it!",link:"Learn more",message:"This website uses Cookies to improve your experience."},enable:!0,palette:{button:{background:"#f0f0f0"},popup:{background:"#1aa3ff"}},theme:"edgeless"},search:{algoliaAppID:"MFXX0N7AVC",algoliaIndex:"losloslos",algoliaSearchKey:"6c722b68c0870b94dd1370d28e984b2f",highlightTag:"em",maxResultLength:10,noResultsFound:"No results found",snippetLength:30,type:"algolia"}}</script><script type=text/javascript src=/js/theme.min.js></script></body></html>